### ISSUE TITLE
install.html not working

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6923]
Migrated from [rt.perl.org#24493](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=24493) (status was 'resolved')

Searchable as RT24493$

**Comment by p5pRT** [comment]
### From warrend@tek.com

#### Created by warrend@tek.com



In perl\-5\.8\.1 I tried the install\.html target\. It started to work and then
had the error shown below\.

It failed here in installhtml\. $^X is \./perl but the current dir is in pod/perlfunc where
there is no perl


415==>          system\($^X\, $splitter\, "\.\./$filename"\) &&
416                 warn "$0&#8203;: error running '$splitter \.\./$filename'"
417                      \." from $podroot/$dirname";

I put links from perlfunc and perlipc back up to \.\./\.\./perl and it then
worked\. That is until it got to doing the index files\.

There it tried to do the destination perlfunc directory which wasn't there
since the source perlfunc directory was empty\. It bombed out because it did
an open on a directory that wasn't there\.

I am not sure if perlfunc should have been empty or the code is not quite
right\.


\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*


LD\_LIBRARY\_PATH=/proj/wdtold/warrend/gnusrc/900180\+perl\-5\.8\.1/perl\-5\.8\.1\_sun5\.8&#8203;:/tools/eda/ldv/bin&#8203;:/proj/eda/ldv/ldv/cur/tools/inca/lib&#8203;:/proj/eda/ldv/ldv/cur/tools/lib \./perl installhtml                   \\
&nbsp;      \-\-podroot=\. \-\-podpath=\. \-\-recurse  \\
&nbsp;      \-\-htmldir=/proj/wdt/gnu\_sun5\.8/perl\-5\.8\.1/lib/5\.8\.1/html     \\
&nbsp;      \-\-htmlroot=/proj/wdt/gnu\_sun5\.8/perl\-5\.8\.1/lib/5\.8\.1/html    \\
&nbsp;      \-\-splithead=pod/perlipc     \\
&nbsp;      \-\-splititem=pod/perlfunc    \\
&nbsp;      \-\-libpods=perlfunc&#8203;:perlguts&#8203;:perlvar&#8203;:perlrun&#8203;:perlop \\
&nbsp;      \-\-verbose
splitting files by head\.
splitting \./pod/perlipc\.pod
&nbsp;        creating \./pod/perlipc/name\.pod
&nbsp;        creating \./pod/perlipc/description\.pod
&nbsp;        creating \./pod/perlipc/signals\.pod
&nbsp;        creating \./pod/perlipc/handling\_the\_sighup\_signal\_in\_daemons\.pod
&nbsp;        creating \./pod/perlipc/named\_pipes\.pod
&nbsp;        creating \./pod/perlipc/deferred\_signals\_\_safe\_signals\_\.pod
&nbsp;        creating \./pod/perlipc/using\_open\_\_\_for\_ipc\.pod
&nbsp;        creating \./pod/perlipc/filehandles\.pod
&nbsp;        creating \./pod/perlipc/background\_processes\.pod
&nbsp;        creating \./pod/perlipc/complete\_dissociation\_of\_child\_from\_parent\.pod
&nbsp;        creating \./pod/perlipc/safe\_pipe\_opens\.pod
&nbsp;        creating \./pod/perlipc/bidirectional\_communication\_with\_another\_process\.pod
&nbsp;        creating \./pod/perlipc/bidirectional\_communication\_with\_yourself\.pod
&nbsp;        creating \./pod/perlipc/sockets\_\_client\_server\_communication\.pod
&nbsp;        creating \./pod/perlipc/internet\_line\_terminators\.pod
&nbsp;        creating \./pod/perlipc/internet\_tcp\_clients\_and\_servers\.pod
&nbsp;        creating \./pod/perlipc/unixdomain\_tcp\_clients\_and\_servers\.pod
&nbsp;        creating \./pod/perlipc/tcp\_clients\_with\_io\_\_socket\.pod
&nbsp;        creating \./pod/perlipc/a\_simple\_client\.pod
&nbsp;        creating \./pod/perlipc/a\_webget\_client\.pod
&nbsp;        creating \./pod/perlipc/interactive\_client\_with\_io\_\_socket\.pod
&nbsp;        creating \./pod/perlipc/tcp\_servers\_with\_io\_\_socket\.pod
&nbsp;        creating \./pod/perlipc/udp\_\_message\_passing\.pod
&nbsp;        creating \./pod/perlipc/sysv\_ipc\.pod
&nbsp;        creating \./pod/perlipc/notes\.pod
&nbsp;        creating \./pod/perlipc/bugs\.pod
&nbsp;        creating \./pod/perlipc/author\.pod
&nbsp;        creating \./pod/perlipc/see\_also\.pod
splitting files by item\.
Can't exec "\./perl"&#8203;: No such file or directory at installhtml line 415\.
installhtml&#8203;: error running '/proj/wdtold/warrend/gnusrc/900180\+perl\-5\.8\.1/perl\-5\.8\.1\_sun5\.8/\./pod/splitpod \.\./perlfunc\.pod' from \./pod/perlfunc at installhtml line 415\.
\./\./ext/attrs/attrs\.pm => /proj/wdt/gnu\_sun5\.8/perl\-5\.8\.1/lib/5\.8\.1/html/\./ext/attrs/attrs\.html
installhtml&#8203;: error running pod/pod2html&#8203;: No such file or directory

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=low

Site configuration information for perl v5.8.1:

Configured by warrend at Tue Nov 11 11:06:44 PST 2003.

Summary of my perl5 (revision 5.0 version 8 subversion 1) configuration:
  Platform:
    osname=solaris, osvers=2.8, archname=sun4-solaris-thread-multi
    uname='sunos rosewood 5.8 generic_108528-18 sun4u sparc sunw,ultra-5_10 solaris '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -fno-strict-aliasing -I/proj/wdt/sun5.8/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O',
    cppflags='-D_REENTRANT -fno-strict-aliasing -I/proj/wdt/sun5.8/include'
    ccversion='', gccversion='3.2.3', gccosandvers='solaris2.8'
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=4321
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' '
    libpth=/usr/lib /usr/ccs/lib /proj/wdt/sun5.8/lib
    libs=-lsocket -lnsl -lgdbm -ldb -ldl -lm -lpthread -lc
    perllibs=-lsocket -lnsl -ldl -lm -lpthread -lc
    libc=/lib/libc.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' -Wl,-E  -z ignore -z lazyload -z combreloc -R /proj/wdt/gnu_sun5.8/perl-5.8.1/lib/5.8.1/sun4-solaris-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags=' -Wl,-E -G -G  -z ignore -z lazyload -z combreloc'

Locally applied patches:
    


@INC for perl v5.8.1:
    /proj/wdt/gnu_sun5.8/perl-5.8.1/lib/5.8.1/sun4-solaris-thread-multi
    /proj/wdt/gnu_sun5.8/perl-5.8.1/lib/5.8.1
    /proj/wdt/gnu_sun5.8/perl-5.8.1/addons/lib/site_perl/5.8.1/sun4-solaris-thread-multi
    /proj/wdt/gnu_sun5.8/perl-5.8.1/addons/lib/site_perl/5.8.1
    /proj/wdt/gnu_sun5.8/perl-5.8.1/addons/lib/site_perl
    .


Environment for perl v5.8.1:
    HOME=/login/warrend
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH=/tools/eda/ldv/bin:/proj/eda/ldv/ldv/cur/tools/inca/lib:/proj/eda/ldv/ldv/cur/tools/lib
    LOGDIR (unset)
    PATH=/stdapps/lsf/5.8/sparcv9/bin:/stdapps/lsf/local/bin:.:/proj/wdt/gnu_sun5.8/perl-5.8.1/bin:/proj/wdt/bin:/proj/wdt/sun5.8/bin:/proj/wdt/X.V11R6/sun5.8/bin:/login/warrend:/login/warrend/bin:/login/warrend/bin/sun5.8:/usr/ucb:/usr/openwin/bin:/usr/ccs/bin:/usr/bin:/etc:/usr/sbin:/proj/dpl/harry/bin:/tools/citrix6.0/wfica:/tools/eda/psd/15.0e/tools/pcb/bin:/tools/eda/psd/15.0e/tools/bin:/tools/eda/psd/15.0e/tools/dfII/bin:/tools/eda/psd/15.0e/tools/fet/bin:/tools/eda/psd/15.0e/tools/specctra/bin:/tools/eda/fetools/psd150/bin:/tools/eda/cadtools/psd150/bin:/tools/eda/ldv/LSF:/tools/eda/ldv/bin:/proj/eda/ldv/ldv/cur/tools/bin:/proj/eda/ldv/ldv/cur/tools/inca/bin:/proj/eda/ldv/lec/cur/bin:/proj/eda/ldv/debussy/cur/bin:/proj/eda/ldv/nLint/cur/bin:/proj/eda/common/vim/cur/bin:/stdapps/cms/bin
    PERL_BADLANG (unset)
    SHELL=/bin/tcsh

```
</details>

**Comment by p5pRT** [comment]
### From @eserte

"warrend@&#8203;tek\.com \(via RT\)" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:

> \# New Ticket Created by  warrend@&#8203;tek\.com 
> \# Please include the string&#8203;:  \[perl \#24493\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt2/Ticket/Display\.html?id=24493 >
> 
> 
> This is a bug report for perl from warrend@&#8203;tek\.com\,
> generated with the help of perlbug 1\.34 running under perl v5\.8\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> In perl\-5\.8\.1 I tried the install\.html target\. It started to work and then
> had the error shown below\.
> 
> It failed here in installhtml\. $^X is \./perl but the current dir is in pod/perlfunc where
> there is no perl
> 
> 
> 415==>          system\($^X\, $splitter\, "\.\./$filename"\) &&
> 416                 warn "$0&#8203;: error running '$splitter \.\./$filename'"
> 417                      \." from $podroot/$dirname";
> 
> I put links from perlfunc and perlipc back up to \.\./\.\./perl and it then
> worked\. That is until it got to doing the index files\.
> 
> There it tried to do the destination perlfunc directory which wasn't there
> since the source perlfunc directory was empty\. It bombed out because it did
> an open on a directory that wasn't there\.
> 
> I am not sure if perlfunc should have been empty or the code is not quite
> right\.
> 

I think the better fix is this patch to installhtml&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- ../bleedperl/installhtml	Wed Apr  2 23:51:39 2003
+++ ./installhtml	Sun Nov 16 20:47:42 2003
@@ -392,6 +392,7 @@ sub split_on_item {
     print "splitting files by item.\n" if $verbose && $#splititem >= 0;
     $pwd = getcwd();
     my $splitter = absolute_path($pwd, "$splitpod/splitpod");
+    my $perl = absolute_path($pwd, $^X);
     foreach my $pod (@splititem) {
 	# figure out the directory to split into
 	$pod      =~ s,^([^/]*)$,/$1,;
@@ -412,7 +413,7 @@ sub split_on_item {
 	    die "$0: error changing to directory $podroot/$dirname: $!\n";
 	die "$splitter not found. Use '-splitpod dir' option.\n"
 	    unless -f $splitter;
-	system($^X, $splitter, "../$filename") &&
+	system($perl, $splitter, "../$filename") &&
 	    warn "$0: error running '$splitter ../$filename'"
 		 ." from $podroot/$dirname";
     }


```
</details>
Regards,
&nbsp;        Slaven

\-\- 
Slaven Rezic \- slaven@&#8203;rezic\.de

&nbsp;    tktimex \- project time manager
&nbsp;    http&#8203;://sourceforge\.net/projects/ptktools/

**Comment by p5pRT** [comment]
### From @rgs

Slaven Rezic wrote&#8203;:
> I think the better fix is this patch to installhtml&#8203;:

Thanks\, applied as \#21739\.

**Comment by p5pRT** [comment]
@rgs - Status changed from 'new' to 'resolved'
